lib/core/src/lib/form/components/widgets/core/form-field.model.ts

```diff
@@ -341,14 +341,27 @@ export class FormFieldModel extends FormWidgetModel {
          but saving back as object: { id: <id>, name: <name> }
          */
         if (json.type === FormFieldTypes.RADIO_BUTTONS) {
+            if (json.value?.options) {
+                this.options = this.parseValidOptions(json.value.options);
+            }
+
             // Activiti has a bug with default radio button value where initial selection passed as `name` value
             // so try resolving current one with a fallback to first entry via name or id
             // TODO: needs to be reported and fixed at Activiti side
-            const entry: FormFieldOption[] = this.options.filter(
+            const matchingOption = this.options.find(
                 (opt) => opt.id === value || opt.name === value || (value && (opt.id === value.id || opt.name === value.name))
             );
 
-            return entry.length > 0 ? entry[0].id : value;
+            if (matchingOption) {
+                return matchingOption.id;
+            }
+
+            if (this.isValidOption(value)) {
+                this.addOption({ id: value.id, name: value.name });
+                return value.id;
+            }
+
+            return value;
```

Comment:
It's 8 years old code (some Activity comments there which made it more unclear what we want to achieve) and I'm not sure what was the reason of returning just value and not caring about its type. After investigation (spent more then month of delving into dropdown/radio buttons widgets), can say they this type should be **string** representing selected **id or name/label** (some comments suggest it).

Part of **parseValue** method only called only once on constructor:
![image](https://github.com/user-attachments/assets/a7117e99-907e-49b3-92cd-3b5596610199)

What we can suppose as well, we want to keep this value (id or name of selected option) in case we will receive full list of options later e.g. from rest api call

Reply:
Indeed our current UI for Radio Buttons Widget is prepared for **selected option id** (just string) but from requested json we can receive:
- selected option **id (string)**
- selected option **name (string)**
- full object of selected option **object { id: string, name: string }**
- full object of selected option + options **{ id: string, name: string, options: [{ id: string, name: string }, { ... }] } -> when my PR will be merged**

Based on what I know so far.

Reply:
I grater detail we do a lots of checks in the scope of this function: 
```
function parseValue(json: any): any {
    if (json.type === FormFieldTypes.DROPDOWN) { return parsedValueForDropdown }
    if (json.type === FormFieldTypes.RADIO_BUTTONS) { return parsedValueForRadio }
    if (this.isDateField(json) || this.isDateTimeField(json)) { return paredValueForDate } 
    if (this.isCheckboxField(json)) { return parsedValueForCheckbox }
}
```
For sure this code should be split e.g. 
`if (json.type === FormFieldTypes.DROPDOWN) { return parseDropdownValue(json.value) }`

where

`function parseDropdownValue(value: any): ParsedDropdownValue { ... }`

with this in place we could improve parseValue function as follows:
```
function parseValue(json: any): ParsedDropdownValue | ParsedRadioValue | ... {
    if (json.type === FormFieldTypes.DROPDOWN) { return parseDropdownValue(json.value) }
    if (json.type === FormFieldTypes.DROPDOWN) { return parseRadioValue(json.value) }
    ...
}
```

But for now I won't cluttered the code with these changes, nonetheless it's a good point in regards of refactoring.